proposals/configuration-level-resources.md

```diff
@@ -0,0 +1,150 @@
+
+```yaml
+  resources:
+    requests:
+      akri.sh/onvif-camera: "2"
+```
+
+## Terms
+
+| Term | Definition |
+| --- | --- |
+| `capacity`  | Maximum amount of containers that can use a device. Defines the number of usage slots a device's device plugin advertises to the kubelet |
+| Configuration-level (CL) resource | An extended resource that is created by the Agent for each Configuration. This equates to the creation of a Kubernetes device plugin for each Configuration. It enables requesting any of a set of (Configuration) resources, such as any IP camera, thermometer, etc. Resources are given the Configuration name. A request to use a CL resource is mapped to an IL resource by the Agent. |
+| Instance-level (IL) resource | An extended resource that is created by the Agent for each discovered device (i.e. Instance), such as a specific IP camera, thermometer, etc. This equates to the creation of a Kubernetes device plugin for each Instance. Resources are named in the format <Configuration Name>-<HASH>, where the input to the one-way hash is a descriptor of the Instance (and node name for local non-shared devices). each Configuration |
+
+## Implementation
+
+To support Configuration-level resources, the Agent will create an additional device plugin for each
+Configuration (that initiates discovery). No changes are needed to the Controller or Discovery Handlers.
+
+The device plugin created for each Configuration will contain `capacity` * `number of
+instances` slots. Each slot will map to a "real" slot of an Instance device plugin. For example,
+after deploying a `onvif` Configuration with a `capacity` of `2`, the `NodeSpec` of a node that
+could see both cameras would be:
+
+```yaml
+Capacity:
+  akri.sh/akri-onvif:         4
+  akri.sh/akri-onvif-8120fe:  2
+  akri.sh/akri-onvif-a19705:  2
+```
+
+Now, an operator or the Controller can create a DaemonSet like the following:
+
+```yaml
+apiVersion: "apps/v1"
+kind: DaemonSet
+metadata:
+  name: onvif-broker-daemonset
+spec:
+  selector:
+    matchLabels:
+      name: onvif-broker
+  template:
+    metadata:
+      labels:
+        name: onvif-broker
+    spec:
+      containers:
+      - name: nginx
+        image: "nginx:latest"
+        resources:
+          requests:
+            "akri.sh/akri-onvif": "2"
+          limits:
+            "akri.sh/akri-onvif": "2"
+```
+
+The Kubernetes scheduler will deploy a Pod to each Node. Pods will only be successfully scheduled to
+a Node and run if the resources exists and are available. Otherwise they will be left in a `Pending`
+state. See this in action by walking through the [note on `Pending` Pods](#Note-on-Pending-Pods).
+
+**Note: More `Pending` Pods is a downside to Configuration-level resources, whether the Deployments are deployed by the Akri Controller or the user. Since
+However, `Pending` Pods only only equates to just more YAML it etcd.**
```

Comment:
can you elaborate more on this?

Reply:
on pending pods? I try to illustrate this point in the #Note-on-pending-pods section. Do you mean explain why they go into a pending state?

Reply:
I just didn't understand the sentence "However, `Pending` Pods only only equates to just more YAML it etcd.**"

Reply:
This is just to say that a `Pending` Pod isnt using up cluster resources. I can clarify this with better wording -- or maybe just adding this sentence.